src/Microsoft.VisualStudio.Debugger.Lua/AD7.Impl/AD7Engine.cs

```diff
@@ -0,0 +1,750 @@
+                (uint)((luaDetoursDllName.Length + 1) * Marshal.SizeOf(typeof(char))),
+                DLLInjector.MEM_COMMIT | DLLInjector.MEM_RESERVE, DLLInjector.PAGE_READWRITE);
+
+            UIntPtr bytesWritten1;
+            DLLInjector.WriteProcessMemory(processHandle, allocMemAddress1,
+                Encoding.Default.GetBytes(luaDetoursDllName),
+                (uint)((luaDetoursDllName.Length + 1) * Marshal.SizeOf(typeof(char))), out bytesWritten1);
+            IntPtr hRemoteThread1 = DLLInjector.CreateRemoteThread(processHandle, IntPtr.Zero, 0, loadLibAddr, allocMemAddress1, 0, IntPtr.Zero);
+
+            IntPtr[] handles1 = new IntPtr[] { hRemoteThread1 };
+            uint index1;
+            CoWaitForMultipleHandles(0, -1, handles1.Length, handles1, out index1);
+
+            string debugDllName = Path.Combine(VS140ExtensionPath, "LuaDebug32.dll");
+
+            IntPtr allocMemAddress2 = DLLInjector.VirtualAllocEx(processHandle, IntPtr.Zero,
+                (uint)((debugDllName.Length + 1) * Marshal.SizeOf(typeof(char))),
+                DLLInjector.MEM_COMMIT | DLLInjector.MEM_RESERVE, DLLInjector.PAGE_READWRITE);
+
+            UIntPtr bytesWritten2;
+            DLLInjector.WriteProcessMemory(processHandle, allocMemAddress2,
+                Encoding.Default.GetBytes(debugDllName), (uint)((debugDllName.Length + 1) * Marshal.SizeOf(typeof(char))), out bytesWritten2);
+
+            IntPtr hRemoteThread2 = DLLInjector.CreateRemoteThread(processHandle, IntPtr.Zero, 0, loadLibAddr, allocMemAddress2, 0, IntPtr.Zero);
+            IntPtr[] handles = new IntPtr[] { hRemoteThread2 };
+            uint index2;
+            CoWaitForMultipleHandles(0, -1, handles.Length, handles, out index2);
+
+
+            AD_PROCESS_ID adProcessId = new AD_PROCESS_ID();
+            adProcessId.ProcessIdType = (uint)enum_AD_PROCESS_ID.AD_PROCESS_ID_SYSTEM;
+            adProcessId.dwProcessId = pi.dwProcessId;
+
+            EngineUtils.RequireOk(port.GetProcess(adProcessId, out process));
+            debugProcess = process;
+
+            return VSConstants.S_OK;
+        }
+
+        [DllImport("ole32.dll")]
+        static extern int CoWaitForMultipleHandles(uint dwFlags, int dwTimeout, int cHandles, IntPtr[] pHandles, out uint lpdwindex);
+
+        private async Task WriteNamedPipeAsync()
+        {
+            string pipeID = pID.ToString();
+
+            using (NamedPipeServerStream pipeServer = new NamedPipeServerStream("luaPipeW" + pipeID, PipeDirection.Out, 1, PipeTransmissionMode.Message, PipeOptions.Asynchronous))
+            {
+                await Task.Factory.FromAsync((cb, state) => pipeServer.BeginWaitForConnection(cb, state), ar => pipeServer.EndWaitForConnection(ar), null);
+
+                using (StreamWriter pipeWriter = new StreamWriter(pipeServer))
+                {
+                    pipeWriter.AutoFlush = true;
+                    // keepWritePipeOpen = true;
```

Comment:
I believe this should be uncommented, otherwise if you tried to launch multiple times then the write pipe wouldn't be open the second time causing issues.